### PR TITLE
Fix the conn_id missing in output table

### PR DESCRIPTION
**Pull request by sunank200**
# Description
## What is the current behavior?

`conn_id` is missing in output table for transform operator. It should be checked for that

closes: #1152

## What is the new behavior?

- Check if `conn_id` is empty
- Push row count of output table to xcom
- Fixed some flaky test

## Does this introduce a breaking change?
No

### Checklist
- [x] Created tests which fail without the change (if possible)
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

File: python-sdk/src/astro/sql/operators/transform.py [hunk 1]
```diff
@@ -60,6 +60,7 @@ def execute(self, context: Context):
             target_table=self.output_table,
             parameters=self.parameters,
         )
+        context["ti"].xcom_push(key="output_table_row_count", value=str(self.output_table.row_count))
```
Comment (pankajastro) [inline python-sdk/src/astro/sql/operators/transform.py:63]: does this only need for the transform operator only and not for other operators inherited from `BaseSQLDecoratedOperator`?

Comment (pankajastro) [reply python-sdk/src/astro/sql/operators/transform.py:63]: since if the output table is not passed then BaseSQLDecoratedOperator  assign it to Table()

Comment (rajaths010494) [reply python-sdk/src/astro/sql/operators/transform.py:63]: Yes, currently only for this operator open lineage is failing to fetch row_count

Comment (pankajastro) [reply python-sdk/src/astro/sql/operators/transform.py:63]: hmm, I see, look like we do not have lineage for `run_raw_sql`

**Review by pankajastro — APPROVED** [review]
LGTM

**Review by phanikumv — APPROVED** [review]

**Pull request merged**